Tests/powershell-yaml.Tests.ps1

```diff
@@ -38,13 +38,13 @@ InModuleScope $moduleName {
 
         # else; handle hashes and arrays specially:
         if ($expected -is [System.Array]) {
-            if ($got -isnot [System.Array] -or ($expected.Count -ne $got.Count)) {
+            if ( -not (,$got | Get-Member -Name 'Count') -or ($expected.Count -ne $got.Count)) {
```

Comment:
This will match ```hashtables``` as well If I am not mistaken. Both ```[System.Hashtable]``` and ```[System.Array]``` have the ```Count``` member. The distinction is important though. Was the test not passing using the previous condition?